appinventor/appengine/src/com/google/appinventor/server/project/youngandroid/GalleryToken.java

```diff
@@ -0,0 +1,71 @@
+// -*- mode: java; c-basic-offset: 2; -*-
+// Copyright 2020 MIT, All rights reserved
+// Released under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+/**
+ * This module is used to generate and validate "tokens" used to
+ * securely communicate between the new MIT App Inventor Gallery and a
+ * given instance of MIT App Inventor
+ *
+ * tokens are serialized protocol buffers which are then encrypted
+ * with KeyCzar
+ *
+ * As a hack to keep the tokens smaller, the timestamp uses python
+ * timestamps (which are Java timestamps divided by 1000 with January
+ * 1, 2020 substracted "Hackito Ergo Sum"
```

Comment:
I do not want to change this. The token gets encrypted, which means that if it is one byte over the blocksize of the cipher, a whole new block is used. Protocol buffers use an efficient representation for small integers that I am taking advantage of here. Finally, this timestamp is ephemeral. It is never saved to permanent storage where it may be confused. This hack is limited to this code and the corresponding "verifytoken" routine on the Gallery side.

Reply:
Fair enough.